lib/tools.js

```diff
@@ -694,5 +870,202 @@ module.exports = {
                 res.json(data);
             }
         };
+    },
+
+    async generateAPiDocs(routes) {
+        let docs = `
+openapi: 3.0.0
+info:
+    title: WildDuck API
+    description: WildDuck API docs
+    version: 1.0.0
+    contact:
+        url: 'https://github.com/nodemailer/wildduck'
+
+servers:
+    - url: 'https://api.wildduck.email'
+
+tags:
+    - name: Addresses
+    - name: ApplicationPasswords
+    - name: Archive
+      description: Archive includes all deleted messages. Once messages are old enough then these are permanenetly deleted from the archive as well. Until then you can restore the deleted messages.
+    - name: Audit
+      description: 'Auditing allows to monitor an email account. All existing, deleted and new emails are copied to the auditing system. See also https://github.com/nodemailer/wildduck-audit-manager'
+    - name: Authentication
+    - name: Autoreplies
+    - name: Certs
+      description: WildDuck allows to register TLS certificates to be used with SNI connections. These certificates are used by IMAP, POP3, API and SMTP servers when a SNI capable client establishes a TLS connection. This does not apply for MX servers.
+    - name: DKIM
+      description: Whenever an email is sent WildDuck checks if there is a DKIM key registered for the domain name of the sender address and uses it to sign the message.
+    - name: DomainAccess
+      description: Add sender domain names to allowlist (messages are all accepted) or blocklist (messages are sent to Spam folder)
+    - name: DomainAliases
+    - name: Filters
+    - name: Mailboxes
+    - name: Messages
+    - name: Settings
+    - name: Storage
+      description: Storage allows easier attachment handling when composing Draft messages. Instead of uploading the attachmnent with every draft update, you store the attachment to the Storage and then link stored file for the Draft.
+    - name: Submission
+    - name: TwoFactorAuth
+    - name: Users
+    - name: Webhooks\n`;
+        const mapPathToMethods = {}; // map -> {path -> {post -> {}, put -> {}, delete -> {}, get -> {}}}
+
+        for (const routePath in routes) {
+            const route = routes[routePath];
+            const { spec } = route;
+
+            if (spec.exclude) {
+                continue;
+            }
+
+            if (!mapPathToMethods[spec.path]) {
+                mapPathToMethods[spec.path] = {};
+            }
+
+            mapPathToMethods[spec.path][spec.method.toLowerCase()] = {};
+            const methodObj = mapPathToMethods[spec.path][spec.method.toLowerCase()];
```

Comment:
Since you're constructing an openapi operation here (I think?), is methodObj really a clear variable name? I think it would make sense to follow the same naming as the openapi spec in all of this for code readability.